tpu-client-next/tests/connection_workers_scheduler_test.rs

```diff
@@ -588,19 +588,16 @@ async fn test_rate_limiting() {
     scheduler_cancel.cancel();
     let stats = join_scheduler(scheduler_handle).await;
 
-    // we get 2 transactions registered as sent (but not acked) because of how QUIC works
-    // before ratelimiter kicks in.
     assert!(
         stats
             == SendTransactionStatsNonAtomic {
-                successfully_sent: 2,
-                write_error_connection_lost: 2,
+                connection_error_timed_out: 1,
```

Comment:
this is how it should have been - when ratelimit is exceeded, we should get an error.